tests/conftest.py

```diff
@@ -0,0 +1,116 @@
+# conftest.py
+# See: https://docs.pytest.org/en/7.1.x/reference/fixtures.html#conftest-py-sharing-fixtures-across-multiple-files
+"""
+The conftest.py file serves as a means of providing fixtures for an
+entire directory. Fixtures defined in a conftest.py can be used by any
+test in that package without needing to import them (pytest will
+automatically discover them).
+"""
+
+import logging
+import os
+import socket
+from typing import Any, AsyncGenerator, cast, Dict
+
+from pymongo import MongoClient
+from pymongo.errors import ServerSelectionTimeoutError
+import pytest
+import pytest_asyncio
+from rest_tools.client import RestClient
+
+from file_catalog.mongo import Mongo
+from file_catalog.server import create
+
+logger = logging.getLogger(__name__)
+
+MonkeyPatch = pytest.MonkeyPatch
+
+
+@pytest_asyncio.fixture
+async def mongo() -> AsyncGenerator[Mongo, None]:
+    """Provide an instance of the File Catalog's internal MongoDB client."""
+    # setup_function
+    mongo_host = os.environ["TEST_DATABASE_HOST"]
+    mongo_port = int(os.environ["TEST_DATABASE_PORT"])
+
+    # did the user override the default database 'file_catalog'?
+    database_name = "file_catalog"
+    if "TEST_DATABASE_NAME" in os.environ:
+        database_name = os.environ["TEST_DATABASE_NAME"]
+
+    # connect a client to the test MongoDB
+    mongo_url = f"mongodb://{mongo_host}"
+    client: MongoClient[Any] = MongoClient(mongo_url, port=mongo_port, connect=True, serverSelectionTimeoutMS=100)
+    db = client[database_name]
+
+    # dump any existing collections to refresh the database
+    try:
+        for collection in db.list_collection_names():
+            if 'system' not in collection:
+                db.drop_collection(collection)
+    except ServerSelectionTimeoutError:
+        raise Exception("Unable to connect to MongoDB; do you have a MongoDB at TEST_DATABASE_{HOST,PORT}?")
+
+    # create an instance of the File Catalog's internal MongoDB client
+    fc_mongo = Mongo(host=mongo_host,
+                     port=mongo_port,
+                     authSource="admin")
+    await fc_mongo.create_indexes()
+
+    try:
+        # provide the client as the fixture
+        yield fc_mongo
+
+    # -------------------------------------------------------------------------
+    # NOTE: *Your Unit Test Function Runs Here*
+    # -------------------------------------------------------------------------
+
+    # teardown_function
+    finally:
+        fc_mongo.close_me.close()
```

Comment:
I didn't catch this earlier. Would a contxetmanager pattern be doable for this instance variable?